metrics/src/test/java/fk/prof/metrics/MetricsTest.java

```diff
@@ -0,0 +1,21 @@
+package fk.prof.metrics;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class MetricsTest {
+  @Test
+  public void testLoadOfMetricNameClassToEnsureNonDuplicatesOfMetricNames() {
+    // This statement will cause class loader to load metricname enum if not already loaded.
+    // Static constructor is invoked which will throw error if duplicate metrics are configured
+    MetricName[] values = MetricName.values();
+    Assert.assertTrue(true);
+  }
+
+  @Test
+  public void testTagEncodingWithCommonNonSpecialChars() {
+    String actual = new ProcessGroupTag("a.b", "c_d", "e-f").toString();
+    String expected = "pgt.a.2Eb_c.5Fd_e.2Df";
```

Comment:
I would need some clarity here. How will this particular encoding work?
For this example, the jmx bean name will be something like "metrics.pgt.a.2Eb.c.5Fd.e.2Df"  (all special chars replaced by dots). So how will we be able to separate the app id from cluster id?